src/Microsoft.AspNet.Mvc.ViewFeatures/Controller.cs

```diff
@@ -1181,6 +1183,128 @@ public virtual CreatedAtRouteResult CreatedAtRoute(string routeName, object rout
+        [NonAction]
+        public virtual ChallengeResult Challenge(IList<string> authenticationSchemes)
+            => new ChallengeResult(authenticationSchemes);
+
+        /// <summary>
+        /// Creates a <see cref="ChallengeResult"/> with the specified <paramref name="properties" />.
+        /// </summary>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
+        /// challenge.</param>
+        /// <returns>The created <see cref="ChallengeResult"/> for the response.</returns>
+        [NonAction]
+        public virtual ChallengeResult Challenge(AuthenticationProperties properties)
+            => new ChallengeResult(properties);
+
+        /// <summary>
+        /// Creates a <see cref="ChallengeResult"/> with the specified specified authentication scheme and
+        /// <paramref name="properties" />.
+        /// </summary>
+        /// <param name="authenticationScheme">The authentication scheme to challenge.</param>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
+        /// challenge.</param>
+        /// <returns>The created <see cref="ChallengeResult"/> for the response.</returns>
+        [NonAction]
+        public virtual ChallengeResult Challenge(string authenticationScheme, AuthenticationProperties properties)
+            => new ChallengeResult(authenticationScheme, properties);
+
+        /// <summary>
+        /// Creates a <see cref="ChallengeResult"/> with the specified specified authentication schemes and
+        /// <paramref name="properties" />.
+        /// </summary>
+        /// <param name="authenticationScheme">The authentication schemes to challenge.</param>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
+        /// challenge.</param>
+        /// <returns>The created <see cref="ChallengeResult"/> for the response.</returns>
+        [NonAction]
+        public virtual ChallengeResult Challenge(
+            IList<string> authenticationSchemes,
+            AuthenticationProperties properties)
+            => new ChallengeResult(authenticationSchemes, properties);
+
+        /// <summary>
+        /// Creates a <see cref="ForbiddenResult"/>.
+        /// </summary>
+        /// <returns>The created <see cref="ForbiddenResult"/> for the response.</returns>
+        [NonAction]
+        public virtual ForbiddenResult Forbid()
+            => new ForbiddenResult();
+
+        /// <summary>
+        /// Creates a <see cref="ForbiddenResult"/> with the specified authentication scheme.
+        /// </summary>
+        /// <param name="authenticationScheme">The authentication scheme to challenge.</param>
+        /// <returns>The created <see cref="ForbiddenResult"/> for the response.</returns>
+        [NonAction]
+        public virtual ForbiddenResult Forbid(string authenticationScheme)
```

Comment:
These ones are more about the security operation and less about the HTTP status code. They do *also* set the status code, but it's far more than that. E.g. the Challenge factory doesn't return a Challenge status code (because there isn't any status code with that name).